### DISCUSSION TITLE
An error notification persists despite the field being filled with a value.

### DISCUSSION THREAD

**Discussion by 6vvvvvv** [react-hook-form/react-hook-form#11219] [Q&A] [answered]
Hi all,

I require assistance and would appreciate some suggestions on this matter. 

Essentially, I would like to receive a notification when a required field is left empty upon form submission. However, the form continues to indicate an issue on field **Logo** even when I have filled in the required value.

Before upload:
![123](https://github.com/react-hook-form/react-hook-form/assets/47306004/e683c1da-8780-40ed-9fd5-79a48e3e7f8b)

After upload and submit
![1234](https://github.com/react-hook-form/react-hook-form/assets/47306004/d29389bb-049e-4ec4-a0e7-1e35c777b11b)

Console log
![2345](https://github.com/react-hook-form/react-hook-form/assets/47306004/e78504bf-d149-4b5d-b52b-188d1c5d7e0d)

Here is the code for the <Logo> field.
![code](https://github.com/react-hook-form/react-hook-form/assets/47306004/058d821e-b5b1-4056-b51f-df74dabf08fd)

Thanks for your help!

**Comment by Moshyfawn** [comment]
I don't see why the validation should fail in the screenshot you provided. The only odd thing I notice is `key={inputKey}` on the file input, which might have something to do with the component being rendered vs. what RHF sees as filled input. Try removing the `key` prop and see if that helps.

**Reply by 6vvvvvv** [reply]
Still persist. It's so weird that it only happens to this field.

**Reply by 6vvvvvv** [reply]
It's interesting that I no longer encounter errors with <Logo> when using an input with a controller. I might consider altering my approach to achieve this. Nevertheless, I appreciate your suggestion. 

RHF team is consistently responsive whenever I seek assistance. Thank you.

**Reply by Moshyfawn** [reply]
What `mode` and `reValidateMode` do you use on your form?

**Reply by 6vvvvvv** [reply]
I only use criteriaMode: "firstError" / shouldFocusError: true  as option.

**Reply by Moshyfawn** [reply, marked as answer]
I overlooked another aspect of how your file input is connected to RHF. You are spreading the return values of the `register` method and subsequently replacing the `onChange` input callback with your custom `handleFileInputChange`. This might be the reason for th behavior you're seeing. To prevent overriding the `onChange` callback from the returned `register` values, use the `onChange` callback prop in the second `options` argument of the `register` method.

```js
{...register("fieldName", { onChange: handleFileInputChange })}
```

**Reply by 6vvvvvv** [reply]
It's fine, thanks.
